PMM-7769 Release Notes 2.16.0 (#433)
percona/pmm-doc: docs/release-notes/2.16.0.md

# Percona Monitoring and Management 2.16.0

<table class="docutils field-list" frame="void" rules="none">
  <colgroup>
    <col class="field-name">
    <col class="field-body">
  </colgroup>
  <tbody valign="top">
    <tr class="field-odd field">
      <th class="field-name">Date:</th>
      <td class="field-body">April 6, 2021</td>
    </tr>
    <tr class="field-even field">
      <th class="field-name">Installation:</th>
      <td class="field-body">
        <a class="reference external" href="https://www.percona.com/software/pmm/quickstart">Installing Percona Monitoring and Management</a></td>
    </tr>
  </tbody>
</table>

Percona Monitoring and Management (PMM) is a free and open-source platform for managing and monitoring MySQL, MongoDB, and PostgreSQL performance.

## Release Highlights

* **Amazon RDS PostgreSQL monitoring**

>     AWS monitoring in PMM now covers PostgreSQL RDS and PostgreSQL Aurora types. PMM will include them in [Discovery UI](../setting-up/client/aws.md#adding-an-amazon-rds-postgresql-instance) and if added after discovery there will be some node related metrics for PostgreSQL databases. Before this release, this was available only to MySQL-related instances from Amazon RDS.

Reviewer: ```suggestion
    AWS monitoring in PMM now covers PostgreSQL RDS and PostgreSQL Aurora types. PMM will include them in [Discovery UI](../setting-up/client/aws.md#adding-an-amazon-rds-postgresql-instance) where they can be added which will result in node related metrics for in addition to PostgreSQL database performance metrics. Before this release, this was available only to MySQL-related instances from Amazon RDS.
```

* **Support for pg_stat_monitor 0.8**

>     Added compatibility with [pg_stat_monitor plugin v 0.8.0](https://github.com/percona/pg_stat_monitor/releases/tag/REL0_8_0_STABLE ). This is not exposing the new features for the plugin in PMM, but just makes Query Analytics metrics collected on the same level it was with version 0.6.0 of the plugin.

Reviewer: ```suggestion
    Added compatibility with [pg_stat_monitor plugin v 0.8.0](https://github.com/percona/pg_stat_monitor/releases/tag/REL0_8_0_STABLE ). This is not exposing the new features for the plugin in PMM yet, but ensures Query Analytics metrics are collected to the same degree it was with version 0.6.0 of the plugin.
```

* **Consistent support of Technical Preview Features**

>     Reworked the PMM Settings page to provide an easy-to-use way for people to understand and manage the ability to activate or deactivate technical preview features. We also provide a better definition of what a [Technical Preview](../details/glossary.md#technical-preview) is.

Reviewer: ```suggestion
    Reworked the PMM Settings page to make it clear what features are in Technical Preview vs General Availability (GA) and to simplify activation/deactivation of technical preview features. We also provide a better definition of what a [Technical Preview](../details/glossary.md#technical-preview) is.
```

> * **Make non-dashboard pages part of grafana fork codebase (migration)**

Reviewer: Is this a highlight?

Author: It's a highlight subtitle with no body, presumably a TODO

Reviewer: @rnovikovP should it be there then?  Is this what a user cares about?

Author: @rnovikovP ?

* **Security Threat Tool Scheduling - Manage execution and execution intervals**

>     Security Threat Tool users are now able to control the Security Check execution time intervals for groups of checks and move some checks between groups, and disable some checks if necessary.

Reviewer: ```suggestion
    Security Threat Tool users are now able to control the Security Check execution time intervals for groups of checks and move checks between groups, and disable individual checks if necessary.
```

* **Azure Discovery and Node metrics extraction**

>     **Technical Preview**: PMM will provide the ability to have the same level of support for Azure DB as we have for AWS RDS. You will be available to discover and easily add Azure databases for monitoring by PMM. This feature is available only if you explicitly turn it ON on the Setting page. Turning this feature OFF will not remove added Services from monitoring, it will just hide the ability to Discover and add new Microsoft Azure Services.

Reviewer: ```suggestion
    **Technical Preview**: PMM will provide the ability to have the same level of support for Azure Database as a Service (DBaaS) as we have for AWS's DBaaS (RDS/Aurora on MySQL or Postgres). You will be able to easily discover and add Azure databases for monitoring by PMM complete with node level monitoring. This feature is available only if you explicitly turn it ON on the Setting page. Turning this feature OFF will not remove added Services from monitoring, it will just hide the ability to Discover and add new Microsoft Azure Services.
```

>     **Technical Preview**: PMM will have the same level of support for Microsoft Azure Database as a Service (DBaaS) as we have for AWS's DBaaS (RDS/Aurora on MySQL or Postgres). You will be able to easily discover and add Azure databases for monitoring by PMM complete with node-level monitoring. This feature is available only if you explicitly turn it ON on the Settings page. Turning this feature OFF will not remove added Services from monitoring, it will just hide the ability to Discover and add new Microsoft Azure Services.

Reviewer 2: ```suggestion
    **Technical Preview**: PMM will have the same level of support for Microsoft Azure Database as a Service (DBaaS) as we have for AWS's DBaaS (RDS/Aurora on MySQL or Postgres). You will be able to easily discover and add Azure databases for monitoring by PMM complete with node-level monitoring. This feature is available only if you explicitly turn it ON on the Settings page. Turning this feature OFF will not remove added Services from monitoring, it will just hide the ability to Discover and add new Microsoft Azure Services.
    
    This feature technical preview because we releasing it as soon as possible to get some feedback from users. We are expecting to do more work on this feature to make it more API and resource-efficient. 
```

*   **[DBaaS] Resource planning and prediction (Resource calculator)**

    **Technical preview**: While creating a DB cluster a user can see a prediction of the resources this cluster will consume with all components as well as the current total and available resources in the K8s. Users will be warned that if they attempt to create a DB cluster it may be unsuccessful because of available resources in the K8s.

> *   **[DBaaS] PSMDB 1.7.0 operator support **

Reviewer 3: extra space breaks formating

Author: Good spot

## New Features

* [PMM-7313](https://jira.percona.com/browse/PMM-7313), [PMM-7610](https://jira.percona.com/browse/PMM-7610): Ability to discover and monitor Amazon RDS PostgreSQL instances with collecting PostgreSQL and RDS node metrics (Thanks to Daniel Guzman Burgos for reporting this issue).

> * [PMM-7345](https://jira.percona.com/browse/PMM-7345): Expose metrics for all available databases in the monitored PostgreSQL server.

Reviewer: ```suggestion
* [PMM-7345](https://jira.percona.com/browse/PMM-7345): Expose metrics for all available databases on a PMM monitored PostgreSQL server.
```

> * [PMM-7344](https://jira.percona.com/browse/PMM-7344): Update postgres_exporter version to 0.8.0. You can see the list of improvements in exporter in the Changelog file.

Reviewer: ```suggestion
* [PMM-7344](https://jira.percona.com/browse/PMM-7344): Update postgres_exporter version to 0.8.0. You can see the list of improvements in exporter in the [Changelog](https://github.com/prometheus-community/postgres_exporter/releases/tag/v0.8.0).
```

> * [PMM-7767](https://jira.percona.com/browse/PMM-7767), [PMM-7696](https://jira.percona.com/browse/PMM-7696): Implement feature flag to enable Microsoft Azure monitoring. Users can [use UI](../setting-up/client/azure.md) or Environment Variable (**NAME**) to pass it during the container creation.

Reviewer: ```suggestion
* [PMM-7767](https://jira.percona.com/browse/PMM-7767), [PMM-7696](https://jira.percona.com/browse/PMM-7696): Implement feature flag to enable Microsoft Azure monitoring. Users can [use UI](../setting-up/client/azure.md) or Environment Variable (`ENABLE_AZUREDISCOVER=1`) to pass it during the container creation.
```

* [PMM-7451](https://jira.percona.com/browse/PMM-7451), [PMM-7337](https://jira.percona.com/browse/PMM-7337): Security Threat Tool: Ability to change intervals for Security checks on the Settings page

* [PMM-7772](https://jira.percona.com/browse/PMM-7772), [PMM-7338](https://jira.percona.com/browse/PMM-7338): Security Threat Tool: Ability to change default execution interval per check

* [PMM-7336](https://jira.percona.com/browse/PMM-7336): Security Threat Tool: Execute checks based on execution interval they belong to

* [PMM-7335](https://jira.percona.com/browse/PMM-7335): Security Threat Tool: Ship security check files with predefined execution interval

* [PMM-7748](https://jira.percona.com/browse/PMM-7748): Add an additional experimental menu for Dashboards on the left side panel

* [PMM-7688](https://jira.percona.com/browse/PMM-7688): Unify UX and layout of all PMM specific pages like Settings, Add Instance etc,

* [PMM-7687](https://jira.percona.com/browse/PMM-7687): Modify links in menus to ensure both menus are working as expected after dashboard URL change

> * [PMM-7705](https://jira.percona.com/browse/PMM-7705): Consistent support of Technical Preview Features

Reviewer: ```suggestion
* [PMM-7705](https://jira.percona.com/browse/PMM-7705): Simplify display of features in technical preview to easily identify them and their current state
```

* [PMM-7521](https://jira.percona.com/browse/PMM-7521), [PMM-7510](https://jira.percona.com/browse/PMM-7510): Integrated Alerting: Improve Alert Rule Templates UX by  Pagination on Rule Templates list

* [PMM-7652](https://jira.percona.com/browse/PMM-7652), [PMM-7674](https://jira.percona.com/browse/PMM-7674), [PMM-7503](https://jira.percona.com/browse/PMM-7503), [PMM-7486](https://jira.percona.com/browse/PMM-7486): DBaaS: While creating the DB cluster  see all and available resources in the K8s cluster, such as Disk, CPU & Memory

* [PMM-7508](https://jira.percona.com/browse/PMM-7508), [PMM-7488](https://jira.percona.com/browse/PMM-7488): DBaaS: See predicted resource usage for selected DB Cluster configuration

* [PMM-7364](https://jira.percona.com/browse/PMM-7364): DBaaS: Show warning before starting creating the cluster if there are not enough resources in the K8s cluster to create DB Cluster with requested configuration

* [PMM-7580](https://jira.percona.com/browse/PMM-7580), [PMM-7359](https://jira.percona.com/browse/PMM-7359): DBaaS: Users can select the database version to use during DB Cluster creation

## Improvements

> * [PMM-7506](https://jira.percona.com/browse/PMM-7506): Security Threat Tool: Exclude PXC internal user accounts from mysql_empty_password check

Reviewer: Ummm...do we want to disclose this?  if it's a security threat to have an unset password how is it less of a threat to have blank passwords in DBaaS?  

Is there better way to say "Do as I say, not as I do"?  How about 'Exclude DBaaS created DB's from certain checks'

Reviewer 2: This is not DBaaS-related, it's  PXC in general. To reduce false-positive STT alerts

Reviewer: Then we should probably word it that way...this looks like we are giving ourselves a pass on a security check when it's the nature of how PXC works...so instead I'd recommend we say "Reduce False Positive's for PXC's internal user account"

> * [PMM-7506](https://jira.percona.com/browse/PMM-7506): Security Threat Tool: Exclude PXC internal user accounts from mysql_empty_password check

Reviewer: ```suggestion
* [PMM-7506](https://jira.percona.com/browse/PMM-7506): Security Threat Tool: Reduce False Positive's Roles automatically created in PXC with no password but cannot be used to login
```

* [PMM-7434](https://jira.percona.com/browse/PMM-7434): Integrated Alerting: Unknown parameters [threshold] error during Add/Update Alert Rule.

* [PMM-7231](https://jira.percona.com/browse/PMM-7231): Integrated Alerting: Disabling channels does nothing

* [PMM-7379](https://jira.percona.com/browse/PMM-7379): Integrated Alerting: Can not edit Alert Rule Name through API

* [PMM-7232](https://jira.percona.com/browse/PMM-7232): Integrated Alerting: Disabling IA does not disable rules evaluation and notifications sending

* [PMM-7119](https://jira.percona.com/browse/PMM-7119): Integrated Alerting: Update error notification for adding/update Alert rule Template – There was inconsistent behavior if you tried to add a new Rule Template with an already-used name.

* [PMM-7543](https://jira.percona.com/browse/PMM-7543): Integrated Alerting: selected section disappears from a breadcrumb after clicking the tab for a second time

* [PMM-7766](https://jira.percona.com/browse/PMM-7766): DBaaS: PMM Upgrade breaks DbaaS get credentials method

* [PMM-7351](https://jira.percona.com/browse/PMM-7351): DBaaS: Safari does not accept float numbers as a custom option in the “Create Cluster” dialogue

> * [PMM-7701](https://jira.percona.com/browse/PMM-7701): DBaaS: PSMDB clusters stuck in initializing due to special characters in secrets

Reviewer: ```suggestion
* [PMM-7701](https://jira.percona.com/browse/PMM-7701): DBaaS: PSMDB clusters stuck in initializing due to special characters in secrets

* [PMM-7728](https://jira.percona.com/browse/PMM-7728): Document add_node for RDS in swagger
```

## Release Highlights

* **Amazon RDS PostgreSQL monitoring**

    AWS monitoring in PMM now covers PostgreSQL RDS and PostgreSQL Aurora types. PMM will include them in [Discovery UI](../setting-up/client/aws.md#adding-an-amazon-rds-postgresql-instance) where they can be added which will result in node related metrics as well as PostgreSQL database performance metrics. Before this release, this was available only to MySQL-related instances from Amazon RDS.

* **Support for pg_stat_monitor 0.8**

    Added compatibility with [pg_stat_monitor plugin v 0.8.0](https://github.com/percona/pg_stat_monitor/releases/tag/REL0_8_0_STABLE ). This is not exposing the new features for the plugin in PMM yet, but ensures Query Analytics metrics are collected to the same degree it was with version 0.6.0 of the plugin.

* **Consistent support of Technical Preview Features**

    Reworked the PMM Settings page to make it clear what features are in Technical Preview vs General Availability (GA) and to simplify activation/deactivation of technical preview features. We also provide a better definition of what a [Technical Preview](../details/glossary.md#technical-preview) is.

> * **Make non-dashboard pages part of grafana fork codebase (migration)**

Reviewer 2: ```suggestion
* **Migration of Settings and other service pages in PMM from Grafana dashboards**

PMM Settings mage and others like Add instance or Inventory will be converted to grafana pages and will not be longer presented as a dashboard. 
Whoever they'll be presented in the left sidebar menu together with other PMM features like DBaaS or Security Checks  

```

    
    This is a feature technical preview because we are releasing it as soon as possible to get some feedback from users. We are expecting to do more work on this feature to make it more API and resource-efficient. 

* **[DBaaS] Resource planning and prediction (Resource calculator)**

    **Technical preview**: While creating a DB cluster a user can see a prediction of the resources this cluster will consume with all components as well as the current total and available resources in the K8s. Users will be warned that if they attempt to create a DB cluster it may be unsuccessful because of available resources in the K8s.

* **[DBaaS] PSMDB 1.7.0 operator support**

> 

Reviewer 2: ```suggestion
DBaaS in PMM will be using recently released    [Percona Kubernetes Operator for Percona Server for MongoDB 1.7.0](https://www.percona.com/doc/kubernetes-operator-for-psmongodb/RN/Kubernetes-Operator-for-PSMONGODB-RN1.7.0.html) to create MongoDB clusters
```

    AWS monitoring in PMM now covers PostgreSQL RDS and PostgreSQL Aurora types. PMM will include them in [Discovery UI](../setting-up/client/aws.md#adding-an-amazon-rds-postgresql-instance) where they can be added which will result in node related metrics as well as PostgreSQL database performance metrics. Before this release, this was available only to MySQL-related instances from Amazon RDS.

* **Support for pg_stat_monitor 0.8**

    Added compatibility with [pg_stat_monitor plugin v 0.8.0](https://github.com/percona/pg_stat_monitor/releases/tag/REL0_8_0_STABLE ). This is not exposing the new features for the plugin in PMM yet, but ensures Query Analytics metrics are collected to the same degree it was with version 0.6.0 of the plugin.

* **Consistent support of Technical Preview Features**

    Reworked the PMM Settings page to make it clear what features are in Technical Preview vs General Availability (GA) and to simplify activation/deactivation of technical preview features. We also provide a better definition of what a [Technical Preview](../details/glossary.md#technical-preview) is.

* **Migration of Settings and other service pages in PMM from Grafana dashboards**

> PMM Settings mage and others like Add instance or Inventory will be converted to grafana pages and will not be longer presented as a dashboard. 

Reviewer 3: paragraph is not aligned for Settings. misses Tab

    Reworked the PMM Settings page to make it clear what features are in Technical Preview vs General Availability (GA) and to simplify activation/deactivation of technical preview features. We also provide a better definition of what a [Technical Preview](../details/glossary.md#technical-preview) is.

* **Migration of Settings and other service pages in PMM from Grafana dashboards**

> PMM Settings mage and others like Add instance or Inventory will be converted to grafana pages and will not be longer presented as a dashboard. 

Reviewer: PMM Settings page and several others (including "Add Instance" and "Inventory" are being converted to Grafana pages and no longer presented as dashboards.  Additionally we're moving the menu to the sidebar navigation for consistency and more flexibility vs the older menu structure.
